cmake/KlepsydraMinimalBuild.cmake

```diff
@@ -0,0 +1,26 @@
+# Copyright 2023 Klepsydra Technologies AG
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Sets minimal build depending on target OS
+macro(set_build_mode)
+    if(${TARGET_OS} STREQUAL ${FREERTOS} OR ${TARGET_OS} STREQUAL ${RTEMS5})
```

Comment:
Remove `TARGET_OS`? (Present in kpsr-core.)

Reply:
since kpsr-core should not have target_os, i guess this whole file can be ignored. Instead each repo should have the cache variable MINIMAL_BUILD set to off by default (setting it to be cache allows it to be overwritten by the command line option).

Reply:
I'm keeping the reduced `KlepsydraMinimalBuild.cmake` so any CE cmake can call `set_build_mode()` as in PE (as opposed to replacing with `set(MINIMAL_BUILD Off...`). @mharshe agree?

Reply:
Hmm in that case it doesn't really make sense, since we have `set` calls elsewhere. So no reason to mask it behind a macro. It makes sense to autoset to off as default when we have non-linux OS because thirdparties (googletest, protobuf..etc) are not available. Here it doesn't make much sense. The only other CE repo is robotics which does not have minimal mode.